composition-js/src/merging/merge.ts

```diff
@@ -1794,7 +1794,9 @@ class Merger {
         if (!sameType(destArg.type!, arg.type!) && !this.isStrictSubtype(arg.type!, destArg.type!)) {
           invalidArgsTypes.add(name);
         }
-        if (destArg.defaultValue !== arg.defaultValue) {
+        const defaultIsArray = Array.isArray(destArg.defaultValue) && Array.isArray(arg.defaultValue);
```

Comment:
Because default values can be any combination of arrays and objects (provided it makes sense for the type), I think this should be something closer to what's in the `mergeDefaultValue()` code (it uses `valueEquals()` after confirming both are not `undefined`).

Reply:
Updated to use `valueEquals()` in [8d73b93](https://github.com/apollographql/federation/pull/3096/commits/8d73b937d2b9ec889854ba1db0c1292535173392)